src/modules/user/services/user.service.ts

```diff
@@ -62,6 +65,37 @@ export class UserService {
       user.password = hashedPassword;
     }
 
-    await this.userRepository.update(id, dtoWithoutPassword);
+    await this.userRepository.update(id, {
+      password: user.password,
+      ...dtoWithoutPassword,
+    });
+  }
+
+  public async getAllUsers(dto: UsersFilterDto) {
+    const { page = 1, limit = 5, order = 'ASC', search = '' } = dto;
+
+    const conditions:
+      | FindOptionsWhere<UserEntity>
+      | FindOptionsWhere<UserEntity[]> = {
+      is_deleted_by_admin: false,
+      user_role: Not(EUserRole.SUPERADMIN),
+      ...(search
+        ? { full_name: Like(`%${search}%`), email: Like(`%${search}%`) }
+        : {}),
+    };
+
+    const [users, totalCount] = await this.userRepository.findAndCount({
+      where: conditions,
+      order: { full_name: order },
+      skip: (page - 1) * limit,
+      take: limit,
+    });
+    return { totalCount, users };
+  }
+
+  public async softDelete(id: string): Promise<void> {
+    const user = await this.isUserExist(id);
```

Comment:
need to check if user was found

Reply:
isUserExist method has this check already